pkg/disruptors/controller_test.go

```diff
@@ -2,63 +2,102 @@ package disruptors
 
 import (
 	"context"
+	"errors"
 	"fmt"
-	"sort"
 	"strings"
 	"testing"
 	"time"
 
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/fake"
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/grafana/xk6-disruptor/pkg/kubernetes/helpers"
 	"github.com/grafana/xk6-disruptor/pkg/testutils/kubernetes/builders"
 )
 
-func Test_InjectAgent(t *testing.T) {
+type fakeCommand struct {
+	err     error
+	exec    []string
+	cleanup []string
+}
+
+func (f fakeCommand) Commands(_ corev1.Pod) ([]string, []string, error) {
+	return f.exec, f.cleanup, f.err
+}
+
+func visitCommands() PodVisitCommand {
+	return fakeCommand{
+		exec:    []string{"command"},
+		cleanup: []string{"cleanup"},
+	}
+}
+
+func Test_VisitPod(t *testing.T) {
```

Comment:
With the current refactor, this is now testing the `PodAgentVisitor` specifically, not any other implementation of `PodVisitor`. Should we rename the test to represent this?